sql/src/main/java/io/crate/expression/scalar/string/InitCapFunction.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.scalar.string;
+
+import io.crate.types.DataTypes;
+import io.crate.expression.scalar.UnaryScalar;
+import io.crate.expression.scalar.ScalarFunctionModule;
+
+import java.util.Arrays;
+import java.util.stream.Stream;
+import java.util.stream.Collectors;
+
+public final class InitCapFunction {
+    private static final String SPLIT_STRING = " ";
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(new UnaryScalar<>("initcap", DataTypes.STRING, DataTypes.STRING, (String val) -> {
+
+            Stream<String> tokenStream = Arrays.stream(val.trim().split(SPLIT_STRING));
+
+            return tokenStream
+                .map(InitCapFunction::toTitleCase)
+                .collect(Collectors.joining(SPLIT_STRING));
+        }));
+    }
+
+    private static String toTitleCase(String word) {
+        if (word.length() == 0)
+            return "";
+
+        char[] chars = word.toCharArray();
+        chars[0] = Character.toUpperCase(chars[0]);
```

Comment:
To support all Unicode characters, we must work with the code point while transforming. Lucene's `CharacterUtils.toUpperCase()` helper function would be taking care of that.